core/src/test/java/com/yahoo/imapnio/async/client/ImapAsyncClientTest.java

```diff
@@ -491,15 +491,15 @@ public void testCreateSessionWithLocalAddressSniSuccessfulSessionDebugOn() throw
         // verify logging messages
         Mockito.verify(logger, Mockito.times(1)).debug(Mockito.eq("[{},{}] connect operationComplete. result={}, imapServerUri={}, sniNames={}"),
                 Mockito.eq(Long.valueOf(2)), Mockito.eq("someUserId"), Mockito.eq("success"), Mockito.eq("imaps://one.two.three.com:993"),
-                Mockito.eq(Arrays.asList("one.two.three.com")));
+                Mockito.eq(Collections.singletonList("one.two.three.com")));
```

Comment:
Any reason Collections.singletonList() is better than Arrays.asList()?

Reply:
It is the reason why we always use "final". singletonList creates a list which doesn't allow any modification with one element. If we have more than one element which doesn't allow modification, we should use toUnmodifiableList.

Reply:
Good! please go through the Arrays.asList and do a test if you call add() on a the list created by Arrays.asList(..)